### DISCUSSION TITLE
SWC/SWF build command

### DISCUSSION THREAD

**Discussion by hydroper** [airsdk/apm#159] [Ideas]
Actually to build SWC/SWF without writting `acompc` or `amxmlc` we've to either:

- Create an Ant script (like `my-project/ant.xml`).
- Create a Batch script (like `my-project/build.cmd`).
- Create a Bash script (like `my-project/build.sh`).
- Create `asconfig.json` and use `asconfigc` from Josh Tynjala.

This is all boring and clusters the project structure. It'd be nice if APM covered the `acompc`/`amxmlc` commands in its `project.apm` or `package.json` file and had a command like `apm build-swc`, which will have, by default:

- `--library-path+=libs`
- `--include-sources+=src`
- `-o=swc/package-id.swc` (or SWF)

Additional semantics:

- When running SWC/SWF build command, automatically run `apm install` if not yet ran.

Additional cases:

- ASDoc?

Package managers such as Cargo, in case of the Rust language, also do a similiar thing.

**Comment by marchbold** [comment]
Agreed. Things like this we are expecting to be added as part of a next round of development. We were first hoping to see how the apm tool was used by the community to get feedback as to the best approach here. 

There are a range of possible solutions we could generate / update and many have asked for IDE integration / setup solutions. Coming from a coding background my approach would be something similar to what you have suggested, but the AIR community is varied so need to consider what would be best for the community I suppose and how best to fit in with common build approaches.

**Comment by hydroper** [comment]
I think that maybe APM could allow optional use of `asconfig.json` from the VSCode's AS&MXML extension (aka. `asconfig`)? If a project has `asconfig.json` and is published to apm.dev, I suppose we could generate API docs for it there via `asdoc` (will just require retrieving the option `compilerOptions.include-sources` from the `asconfig.json` file). I also think Cargo's repository uses a Docker instance to generate docs for safety reasons (because a package can have a `build.rs` script that could compromise their server).

I've just created some `asconfig` templates here: https://github.com/hydroper/adobe-air-templates

**Reply by marchbold** [reply]
I have integrating some of the IDEs in the future scope for apm. If you start a new discussion around your idea we can develop it further.

**Reply by hydroper** [reply]
@marchbold Thanks for quick reply! Right, I may create a new discussion soon.
